### PR TITLE
Updated Quick and Nimble

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

This PR updates the test dependencies [Quick](https://github.com/Quick/Quick/releases/tag/v3.0.0) and [Nimble](https://github.com/Quick/Nimble/releases/tag/v9.0.0) to the latest majors, making it possible to test Auth0.swift with Xcode 12. This update will also allow to start addressing the [compilation](https://github.com/auth0/Auth0.swift/issues/397) [issues](https://github.com/auth0/Auth0.swift/issues/396) on BigSur.

**The new majors of Quick and Nimble no longer support Swift versions < 5.2. This means that the _tests_ can no longer be run on Xcode versions < 11.4. The CircleCI config has been updated accordingly.**

### Testing

* [ ] This change adds unit test coverage
* [X] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [X] All existing and new tests complete without errors
* [ ] All active GitHub checks have passed

### CODE REVIEW

**Comment by cocojoe** [comment]
So this PR will be a breaking change in that Carthage at the very least will no longer work in Xcode versions below 12?
Do CocoaPods still work, I don't think they do any testing?

Although this is necessary for Xcode 12 and Carthage. This really needs to be called out explicitly in the README that for Xcode <12 use Minor X for Xcode >=12 this minor is needed.

**Comment by Widcket** [comment]
@cocojoe Quick and Nimble are dropping support for Xcode < 11.4, so it would still work with Xcode 11.4+. So no, the minimum supported Xcode does not shift to 12.
Why would the samples have to be updated? They use Cocoapods.

**Comment by Widcket** [comment]
@cocojoe besides, the README currently states that the supported Xcode versions are 11.x. So the only thing that would actually change is that to use Carthage you would have to use Xcode 11.4+, instead of 11.0-11.3. And that just because the library uses Carthage to manage its own dependencies. The Podspec does not even mention Quick and Nimble.
But, I agree that the above (about Carthage) should be added to the README.

**Comment by cocojoe** [comment]
This needs a heavy call out and clear context, if the minimum listed is 11.x then give the circumstances I think it's okay. We just need to be very explicit to avoid any potential confusion.

**Comment by Widcket** [comment]
Correction, it says "Xcode 10.x / 11.x" However, you can't submit apps to the AppStore that were built with Xcode 10.x anymore, so it's effectively EOL'd.

**Comment by Widcket** [comment]
And with the current major being 4 years old, Xcode 10 is not the first Xcode version to stop being supported by this library, I suppose.

**Comment by Widcket** [comment]
@cocojoe I've added a notice to the README.

File: README.md [hunk 1]
```diff
@@ -5,13 +5,14 @@
...
 
 Swift toolkit that lets you communicate efficiently with many of the [Auth0 API](https://auth0.com/docs/api/info) functions and enables you to seamlessly integrate the Auth0 login.
 
-## Important Notice
-[Behaviour changes in iOS 13](https://github.com/auth0/Auth0.swift/pull/297) related to Web Authentication require that developers using Xcode 11 with this library **must** compile using Swift 5.x. This *should* be the default setting applied when updating, unless it has been manually set. However, we recommend checking that this value is set correctly.
+## Important Notices
+- [The latest majors of Quick and Nimble](https://github.com/auth0/Auth0.swift/pull/421) require that developers using Carthage **must** use Xcode 11.4+ only, starting from Auth0.swift [1.29.0](https://github.com/auth0/Auth0.swift/releases/tag/1.29.0).
+- [Behaviour changes in iOS 13](https://github.com/auth0/Auth0.swift/pull/297) related to Web Authentication require that developers using Xcode 11 with this library **must** compile using Swift 5.x. This *should* be the default setting applied when updating, unless it has been manually set. However, we recommend checking that this value is set correctly.
```
Comment (cocojoe) [inline README.md:15]: Should we create a historical notices section and move this part to then? iOS 13 feels way less relevant now.

**Review by cocojoe — APPROVED** [review]

**Pull request merged**